### DISCUSSION TITLE
Table renders with outdated data before query finishes running

### DISCUSSION THREAD

**Discussion by rootifera** [ToolJet/ToolJet#11639] [Q&A]
Hello, 

I'm having trouble displaying updates in my table.  

I have a page with a Tab container with 4 tabs. 4th tab is for entering data into my database (through a restapi call).  And I also have a javascript query for getting data from my API. the query runs at application start and populates the table in the first tab. 

I added a "on tab switch" event, which runs the table populating query so after I add new data I'd receive new rows. After I add the new data, when I switch to the first tab to see if new data is there, I don't see them. Only after I switch tabs again or refresh the page the data appears. 

In my opinion the table renders with old data because it's not waiting for table populating query.   In the Loading state I added "{{queries.fillCPUTable.isLoading}}" but that doesn't help, I guess because this was previously loaded (or maybe I don't have anything in my js to tell loading started/finished).

I've been through the documents but I couldn't find anything useful.  My purpose is simple:

1 - Switch to tab4, fill the form, click add.
2 - Switch to tab1, where the table is, see the new data.

Can someone please help? 

Thanks

**Comment by PriteshKiri** [comment]
Hey @rootifera ,

Can you please drop your query in the general channel of our [Slack](https://www.tooljet.com/slack) community?

Our team would be happy to assist you there.
